app/dashboard/models.py

```diff
@@ -2226,6 +2226,26 @@ def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
 
         return params
 
+    @property
+    def locations(self):
+        from app.utils import get_location_from_ip
+        locations = []
+        try:
+            for login in self.actions.filter(action='Login'):
+                if login.location_data:
+                    locations.append(login.location_data)
+                else:
+                    location_data = get_location_from_ip(login.ip_address)
+                    login.location_data = location_data
+                    login.save()
+                    # Update all actions with same ip_address ??
+                    # UserAction.objects.filter(action='Login',ip_address=login.ip_address).update(location_data=location_data)
```

Comment:
This is one of the reasons I raised the concern on the original PR at:  https://github.com/gitcoinco/web/pull/2091/files#r212628357

I agree with the above assessment.  The main reason I mentioned migrating to a `Location` intermediate model was to avoid costly calls to `location_data` via `select_related`, but I'm not opposed to the outlined approach.

Reply:
Sounds good, I've removed the commented code in the latest commit.

Thanks for weighing in.